soc/arm64/intel_socfpga/agilex/mmu_regions.c

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (c) 2021, Intel Corporation. All rights reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <arch/cpu.h>
+#include <arch/arm64/arm_mmu.h>
+
+static const struct arm_mmu_region mmu_regions[] = {
+
+	MMU_REGION_FLAT_ENTRY("PERIPHERAL",
+			      0xF7000000,
+			      0x08E00000,
```

Comment:
What's this huge hard-coded area?

Reply:
This is the peripheral register map range for all peripheral in Agilex. We enable for OS access for initial bring.

Reply:
Uhm, I see two problems here:
1. You are missing at least two device regions here: UART and CLOCK that have entries in the DT (you only have GIC).
2. Why do you need to map the whole peripheral range here? Do you have some kind of application running that needs that?

Reply:
UART and CLOCK register are within this huge peripheral register range, we just enable for all during bring up without specific application.

Agreed that this is a problem.

I will do a clean up to only enable for UART and CLOCK, which means only enable the peripheral that we enabled in the code.

Reply:
Remove huge area of PERIPERAL from MMU table, only include:
(1) SYSTEM_MANAGER and CLOCK register map for clock driver
(2) UART0 register map for uart driver